core/config_utils_test.go

```diff
@@ -226,25 +227,75 @@ func TestGetServicePropertiesFromVCAP(t *testing.T) {
 	clearTestVCAP()
 }
 
-func TestLoadFromVCAPServices(t *testing.T) {
-	vcapServices := `{
-		"watson": [{
-			"credentials": {
-				"url": "https://gateway.watsonplatform.net/compare-comply/api",
-				"username": "bogus username",
-				"password": "bogus password",
-				"apikey": "bogus apikey"
-			}
-		}]
-	}`
-	os.Setenv("VCAP_SERVICES", vcapServices)
-	credential1 := loadFromVCAPServices("watson")
-	assert.Equal(t, "bogus apikey", credential1.APIKey)
+func TestLoadFromVCAPServicesWithServiceEntries(t *testing.T) {
+	data, err := ioutil.ReadFile("../resources/vcap_services.json")
+	isNil := assert.Nil(t, err, "There should be no i/o error.")
+	if !isNil {
+		return
+	}
+	os.Setenv("VCAP_SERVICES", string(data))
+	// Verify we checked service entry names first
+	credential1 := loadFromVCAPServices("service_entry_key_and_key_to_service_entries")
+	isNotNil := assert.NotNil(t, credential1, "Credentials1 should not be nil")
+	if !isNotNil {
+		return
+	}
+	assert.Equal(t, "not-a-username", credential1.Username)
+	assert.Equal(t, "not-a-password", credential1.Password)
+	assert.Equal(t, "https://on.the.toolchainplatform.net/devops-insights/api", credential1.URL)
+	// Verify we checked keys that map to lists of service entries
+	credential2 := loadFromVCAPServices("key_to_service_entry_1")
+	isNotNil = assert.NotNil(t, credential2, "Credentials2 should not be nil")
+	if !isNotNil {
+		return
+	}
+	assert.Equal(t, "not-a-username", credential2.Username)
+	assert.Equal(t, "not-a-password", credential2.Password)
+	assert.Equal(t, "https://on.the.toolchainplatform.net/devops-insights/api", credential2.URL)
+	credential3 := loadFromVCAPServices("key_to_service_entry_2")
+	isNotNil = assert.NotNil(t, credential3, "Credentials3 should not be nil")
+	if !isNotNil {
+		return
+	}
+	assert.Equal(t, "not-a-username-3", credential3.Username)
+	assert.Equal(t, "not-a-password-3", credential3.Password)
+	assert.Equal(t, "https://on.the.toolchainplatform.net/devops-insights-3/api", credential3.URL)
+	os.Unsetenv("VCAP_SERVICES")
+}
+
+func TestLoadFromVCAPServicesEmptyService(t *testing.T) {
+	data, err := ioutil.ReadFile("../resources/vcap_services.json")
+	isNil := assert.Nil(t, err, "There should be no i/o error.")
+	if !isNil {
+		return
+	}
+	os.Setenv("VCAP_SERVICES", string(data))
+	// Verify we checked service entry names first
+	credential := loadFromVCAPServices("empty_service")
+	assert.Nil(t, credential, "Credentials1 should not be nil")
+	os.Unsetenv("VCAP_SERVICES")
+}
+
+func TestLoadFromVCAPServicesNoCredentials(t *testing.T) {
```

Comment:
This testcase looks the same as the method above.  Both are retrieving the "empty_service" entry and checking to make sure a nil credential was returned.   Am I missing something?  :)